examples/reshape.cpp

```diff
@@ -0,0 +1,728 @@
+
+  RAJA::View<double, RAJA::Layout<3, int, 2>> Aview(A, layout1);
+  RAJA::View<double, RAJA::Layout<3, int, 2>> Bview(B, layout1);
+  RAJA::View<double, RAJA::Layout<3, int, 2>> Cview(C, layout1);
+#else
+
+  auto Aview = RAJA::Reshape<RAJA::layout_right>::get(A, N, N_r, N_c);
+  auto Bview = RAJA::Reshape<RAJA::layout_right>::get(B, N, N_r, N_c);
+  auto Cview = RAJA::Reshape<RAJA::layout_right>::get(C, N, N_r, N_c);
+
+#endif
+  // _permutedlayout_defviews_end
+
+//
+// Allocate space for data in layout 2
+//
+  double *A2 = memoryManager::allocate<double>(N_c * N_r * N);
+  double *B2 = memoryManager::allocate<double>(N_c * N_r * N);
+  double *C2 = memoryManager::allocate<double>(N_c * N_r * N);
+
+//
+// Permuted layout - equivalent to indexing using the following macro
+// #define Aview2(e, r, c) A2[e + N*(c + N_c*r)]
+// In this case the element index has unit stride (argument 0).
+//
+  // _permutedlayout_permviews_start
+#if 0
+    std::array<RAJA::idx_t, 3> perm2 {{0, 1, 2}};
+  auto layout2 =
+      RAJA::make_permuted_layout( {{N, N_r, N_c}}, perm2 );
+
+  RAJA::View<double, RAJA::Layout<3, int, 0>> Aview2(A2, layout2);
+  RAJA::View<double, RAJA::Layout<3, int, 0>> Bview2(B2, layout2);
+  RAJA::View<double, RAJA::Layout<3, int, 0>> Cview2(C2, layout2);
+#else
+
+  auto Aview2 = RAJA::Reshape<RAJA::layout_left>::get(A2, N, N_r, N_c);
+  auto Bview2 = RAJA::Reshape<RAJA::layout_left>::get(B2, N, N_r, N_c);
+  auto Cview2 = RAJA::Reshape<RAJA::layout_left>::get(C2, N, N_r, N_c);
+#endif
+  // _permutedlayout_permviews_end
+
+//
+// Initialize data for layout 1 and layout 2 arrays/views.
+//
+// When OpenMP is enabled, we use an OpenMP exec policy for
+// "first touch" initialization.
+//
+#if defined(RAJA_ENABLE_OPENMP)
+  using INIT_POL = RAJA::omp_parallel_for_exec;
+#else
+  using INIT_POL = RAJA::seq_exec;
+#endif
+
+  std::cout<<"populating"<<std::endl;
+  RAJA::forall<INIT_POL>(RAJA::TypedRangeSegment<int>(0, N), [=](int e) {
+    for (int row = 0; row < N_r; ++row) {
+      for (int col = 0; col < N_c; ++col) {
+        Aview(e, row, col) = row;
+        Bview(e, row, col) = col;
+        Cview(e, row, col) = 0;
+
+        Aview2(e, row, col) = row;
+        Bview2(e, row, col) = col;
+        Cview2(e, row, col) = 0;
+      }
+    }
+  });
+  std::cout<<"pop complete"<<std::endl;
+
+
+//----------------------------------------------------------------------------//
+
+  std::cout << " \n Running batched matrix multiplication"
+            << " with layout 1 (RAJA - sequential) ... " << std::endl;
+
+  minRun = std::numeric_limits<double>::max();
+  for (int i = 0; i < NITER; ++i) {
+
+    timer.start();
+    // _permutedlayout_batchedmatmult_loop_start
+    RAJA::forall<RAJA::seq_exec>(RAJA::TypedRangeSegment<int>(0, N),
+      [=](int e) {
+
+        Cview(e, 0, 0) = Aview(e, 0, 0) * Bview(e, 0, 0)
+                         + Aview(e, 0, 1) * Bview(e, 1, 0)
+                         + Aview(e, 0, 2) * Bview(e, 2, 0);
```

Comment:
This spacing does not spark joy.

Reply:
Also, since it's an example, would making it a doubly-nested loop to compute each CView(e, i, j) entry make it clearer and easier to read?